src/App.ts

```diff
@@ -10,17 +10,33 @@ const error = debug('team-link:error');
 
 export default class App {
     private app: express.Express;
-    private server!: Server;
+    private port: string;
+    private host: string;
+    private staticPath: string;
+    private clientPath: string;
+    private server: Server;
 
-    constructor () {
+    constructor ({ port = config.port, host = config.host, staticPath = config.staticPath, clientPath = config.clientPath } = {}) {
         this.app = express();
+        this.port = port;
+        this.host = host;
+        this.staticPath = staticPath;
+        this.clientPath = clientPath;
+        this.server = new Server({ app: this.app, port, host, staticPath, clientPath });
     }
 
-    async createServer ({ port = config.port, host = config.host, staticPath = config.staticPath, clientPath = config.clientPath } = {}): Promise<Server> {
+    static async createServer ({ port = config.port, host = config.host, staticPath = config.staticPath, clientPath = config.clientPath } = {}): Promise<App> {
```

Comment:
We should think about the name of this method. I think it can be just `create` since you create an App instance here.